src/phoenix/server/api/routers/trace_handler.py

```diff
@@ -54,7 +60,122 @@ async def post(self, request: Request) -> Response:
         for resource_spans in req.resource_spans:
             project_name = get_project_name(resource_spans.resource.attributes)
             for scope_span in resource_spans.scope_spans:
-                for span in scope_span.spans:
-                    self.traces.put(decode(span), project_name=project_name)
+                for otlp_span in scope_span.spans:
+                    span = decode(otlp_span)
+                    async with self.db() as session:
+                        await _insert_span(session, span, project_name)
+                    self.traces.put(span, project_name=project_name)
                     await asyncio.sleep(0)
         return Response()
+
+
+async def _insert_span(session: AsyncSession, span: Span, project_name: str) -> None:
+    if not (
+        project_rowid := await session.scalar(
```

Comment:
just a question: Is there a reason why not to use the ORM in this case?

Reply:
i'm updating it. this was copied pasted, so just wanted to make sure it's still working